src/main/java/com/google/devtools/build/lib/remote/RemoteActionFileSystem.java

```diff
@@ -185,14 +185,14 @@ protected PathFragment readSymbolicLink(PathFragment path) throws IOException {
   }
 
   @Override
-  protected void createSymbolicLink(PathFragment linkPath, PathFragment targetFragment)
+  protected void createSymbolicLink(PathFragment linkPath, String rawTarget)
```

Comment:
I am none too happy about turning this into just taking a raw string. What is it about the PathFragment normalization that needs to be addressed in this commit? Passing around simple strings increases the ways things can go wrong.

Reply:
A symlink with target `bar/../foo` is not the same as a symlink with target `foo`: If `bar` is itself a symlink (or a junction on Windows), then `bar/../foo` will resolve relative to the target of `bar`.

Given these complications around symlinks, there has to be a way for Bazel to create "raw" symlinks via its filesystem abstractions. The respective PR is still WIP, but I plan to add comments explaining that the `PathFragment` overload should be preferred in almost all cases - except where it doesn't work, e.g. when explicitly creating a raw symlink.